src/katgpucbf/fgpu/delay.py

```diff
@@ -219,6 +219,7 @@ def _popleft_until(self, start: int) -> None:
                 f"Timestamp {start} is before start of first linear model "
                 f"at {self._models[0].start} - possibly due to non-monotonic queries",
                 NonMonotonicQueryWarning,
+                stacklevel=3,
```

Comment:
I'm curious as to what this warning was.

Reply:
> B028: No explicit stacklevel argument found. The warn method from the warnings module uses a stacklevel of 1 by default. This will only show a stack trace for the line on which the warn method is called. It is therefore recommended to use a stacklevel of 2 or greater to provide more information to the user.

I went with 3 rather than 2 because this is a private helper function rather than the function the user actually calls.